### PR TITLE
ME-1788 fixhomedir Make sure we use the correct homedir

### PR DESCRIPTION
**Pull request by atoonk**
# Description
This fix makes sure we use the correct homedir
if you use sudo, we will make sure to the original users homedir ie. env SUDO_USER

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: cmd/client/db/datagrip.go [hunk 1]
```diff
@@ -89,7 +90,8 @@ var dataGripCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
-		home, err := os.UserHomeDir()
+		home, err := util.GetUserHomeDir()
+
```
Comment (adrianosela) [inline cmd/client/db/datagrip.go:94]: rm newline.

File: internal/client/client.go [hunk 2]
```diff
@@ -354,7 +355,10 @@ func GetResourceInfo(logger *zap.Logger, hostname string) (info ResourceInfo, er
 }
 
 func MTLSTokenFile() string {
-	home := os.Getenv("HOME")
+	home, err := util.GetUserHomeDir()
+	if err != nil {
+		log.Fatal(err)
```
Comment (adrianosela) [inline internal/client/client.go:360]: formatted error -- consider also propagating the error upstream and handling it there

File: internal/client/resources.go [hunk 3]
```diff
@@ -235,7 +236,11 @@ func saveToken(token string) error {
 	if err != nil {
 		return fmt.Errorf("couldn't get currently logged in operating system user: %w", err)
 	}
-	homedir := currentUser.HomeDir
+	homedir, err := osutil.GetUserHomeDir()
+
```
Comment (adrianosela) [inline internal/client/resources.go:240]: rm newline.

File: internal/util/osuser.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"runtime"
+)
+
```
Comment (adrianosela) [inline internal/util/osuser.go:9]: godoc -- what does this do; what is SUDO_USER?

**Review by adrianosela — APPROVED** [review]

File: internal/client/client.go (hunk 2 above)
Comment (atoonk) [reply internal/client/client.go]: made it slightly better

**Pull request merged**